### ISSUE TITLE
Some extra rule inclusion suggestions

### ISSUE THREAD

**Issue by jrfnl** [WordPress-Coding-Standards/WordPress-Coding-Standards#607]
##### For `core` _(or `extra`)_:

To cover: https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#clever-code
- `Squiz.Formatting.DisallowInlineIf`: Stops inline IF statements from being used.
- `Squiz.PHP.DisallowMultipleAssignments`: Ensures that there is only one value assignment on a line, and that it is the first thing on the line.
- [Move from `extra` to `core`] `Generic.Formatting.DisallowMultipleStatements`: Ensures each statement is on a line by itself.
##### For `extra` to further enhance the code analysis related sniffs:
- `Squiz.PHP.NonExecutableCode`: Warns about code that can never been executed. This happens when a function returns before the code, or a break ends execution of a statement etc.
- `Squiz.Operators.IncrementDecrementUsage`: Tests that the ++ operators are used when possible and not used when it makes the code confusing.
- `Squiz.Operators.ValidLogicalOperators`: Checks to ensure that the logical operators 'and' and 'or' are not used. Use the && and || operators instead.
- `Squiz.PHP.DisallowComparisonAssignment`: Ensures that the value of a comparison is not assigned to a variable. **_(probably not refined enough)_**
- One more: `Squiz.Functions.FunctionDuplicateArgument`: Checks that duplicate arguments are not used in function declarations.

Opinions ?

**Comment by GaryJones** [comment]
I'd be happy with that. Perhaps add them early into the next cycle, so we can give them a run (I always use `develop` branch) before release?

**Comment by westonruter** [comment]
Look good, however:

> Squiz.PHP.DisallowComparisonAssignment: Ensures that the value of a comparison is not assigned to a variable. (probably not refined enough)

If I understand this right, it would disallow PHP like:

```php
$is_invalid = ( is_null( $value ) || wp_is_error( $value ) );
```

Is that right? And if so, I very much disagree with that sniff. It is great for readability to store complex conditionals into a variable and then use that variable in `if` statement. This is used in core: https://github.com/xwp/wordpress-develop/blob/e7bd22f87eb3c1edd67636d49ba29a1ddb597c32/src/wp-includes/class-wp-customize-nav-menus.php#L900-L928

**Comment by GaryJones** [comment]
For a conditional with 12 parts and 5 comment lines, I'd say it needs it's own function :-)

Then you're left line 928 as `if ( $this->can_partial_refresh() ) {`.

**Comment by westonruter** [comment]
@GaryJones fair point. Nevertheless, there are at least 10 examples in Core of storing a conditional in a variable:

```
$ ack '\$(is|can|has)_\w+\s=\s.+\s(==|>=|!=|<=|<|>)\s' --php wp-admin/ wp-includes/ | grep -v create_function | cat -
wp-admin/edit-form-advanced.php:583:	$has_sample_permalink = $sample_permalink_html && 'auto-draft' != $post->post_status;
wp-admin/includes/class-ftp.php:165:		$is_windows = ($this->OS_remote == FTP_OS_Windows);
wp-admin/includes/image-edit.php:33:		$can_restore = $backup_sizes['full-orig']['file'] != basename( $meta['file'] );
wp-admin/widgets.php:409:		$is_inactive_widgets = 'wp_inactive_widgets' == $registered_sidebar['id'];
wp-includes/author-template.php:478:		$is_multi_author = 1 < count( $rows ) ? 1 : 0;
wp-includes/class-phpmailer.php:1711:        $is_utf8 = (strtolower($this->CharSet) == 'utf-8');
wp-includes/customize/class-wp-customize-nav-menu-item-setting.php:385:		$is_placeholder = ( $this->post_id < 0 );
wp-includes/customize/class-wp-customize-nav-menu-setting.php:221:		$is_placeholder = ( $this->term_id < 0 );
wp-includes/date.php:406:				$is_between = $_value >= $check['min'] && $_value <= $check['max'];
wp-includes/vars.php:118:$is_iis7 = $is_IIS && intval( substr( $_SERVER['SERVER_SOFTWARE'], strpos( $_SERVER['SERVER_SOFTWARE'], 'Microsoft-IIS/' ) + 14 ) ) >= 7;
```

**Comment by jrfnl** [comment]
@westonruter The current core code base does not comply with quite a few of the standards that have (since) been set and documented in the handbook, so IMHO that is not a reason not to include rules which can be related to the published standard as it is now

**Comment by westonruter** [comment]
In any case, I'm not a fan of `Squiz.PHP.DisallowComparisonAssignment` and I would vote against its inclusion as a core coding standard 😄

**Comment by jrfnl** [comment]
I believe I suggested it for `extra` not `core` :grinning:
